shuup_tests/front/test_checkout.py

```diff
@@ -72,3 +77,102 @@ def test_address_phase_anonymous_user(rf):
     resp = view_func(request)
     assert 'company' in resp.context_data['form'].form_defs
 
+
+# BROWSER TESTS BELOW
+# -------------------
+
+
+def create_orderable_product(name, sku, price):
+    supplier = get_default_supplier()
+    shop = get_default_shop()
+    product = create_product(sku=sku, shop=shop, supplier=supplier, default_price=price, name=name)
+    return product
+
+
+@pytest.mark.browser
+@pytest.mark.djangodb
+def test_browser_checkout(browser, live_server, settings):
+    # initialize
+    product_name = "Test Product"
+    get_default_shop()
+    pm = get_default_payment_method()
+    sm = get_default_shipping_method()
+    create_orderable_product(product_name, "test-123", price=100)
+    OrderStatus.objects.create(
+        identifier="initial",
+        role=OrderStatusRole.INITIAL,
+        name="initial",
+        default=True
+    )
+    browser = initialize_front_browser_test(browser, live_server)
+    assert browser.is_text_present("Welcome to Default!")
+
+    assert browser.is_text_present("Newest Products")
+    assert browser.is_text_present(product_name)
+
+    # open product
+    browser.find_by_id("product-1").click()
+
+    # add product to basket
+    browser.find_by_id("add-to-cart-button").click()
+
+    # TODO: make this check better
+    # assume it was added to cart
+    assert browser.is_text_present("(1)")
+
+    browser.reload()  # reload because of the fancy popover...
+
+    browser.find_by_id("navigation-basket-partial").click()
+    browser.find_link_by_href("/basket/").first.click()
+
+    assert browser.is_text_present("Shopping cart")  # we are in basket page
+    assert browser.is_text_present(product_name)  # product in basket
+
+    # to checkout
+    browser.find_link_by_href("/checkout/").first.click()
+
+    customer_name = "Test Tester"
+    customer_street = "Test Street"
+    customer_city = "Test City"
+    customer_country = "US"
+
+    browser.fill("billing-name", customer_name)
+    browser.fill("billing-street", customer_street)
+    browser.fill("billing-city", customer_city)
+    browser.select("billing-country", customer_country)
+    # browser.execute_script('document.getElementById("same_as_billing").checked=true')  # same as billing
+
+    browser.find_by_css(".btn.btn-primary.btn-lg.pull-right").first.click()  # click "continue"
```

Comment:
Might be possible, not sure. Need to investigate. This is the reason I added the comments :)